core/Error.h

```diff
@@ -157,9 +157,10 @@ class ErrorBuilder {
     }
 
     void addAutocorrect(AutocorrectSuggestion &&autocorrect);
-    template <typename... Args> void replaceWith(Loc loc, const std::string &msg, const Args &... args) {
-        std::string formatted = fmt::format(msg, args...);
-        addAutocorrect(AutocorrectSuggestion(loc, move(formatted)));
+    template <typename... Args>
+    void replaceWith(const std::string &title, Loc loc, const std::string &replacement, const Args &... args) {
```

Comment:
every time you say `const std::string &` in argument poistion, you actually want a `string_view`

Reply:
what about when you say it? :D

![image](https://user-images.githubusercontent.com/4211838/62402986-63983f80-b53f-11e9-906d-0f1b42dc4af0.png)

on a real note, the first two in that screenshot change without issues as well as `replacement`. however, the third (`title`) does not since `AutocorrectSuggestion` wants a `string`. Why does `std::string &` work since it's also a reference? does C++ just auto-insert the copy v.s. with `string_view` where you manually need to wrap it in `string(...)`?

Reply:
With `std::string &`, C++ does implicitly copy the string.

With `string_view`, all copies are explicit and must be wrapped in `string()`.

Reply:
Is there an advantage to using `string_view` + `string()` over just `std::string &`? It seems like no per https://stackoverflow.com/questions/40127965/how-exactly-is-stdstring-view-faster-than-const-stdstring since I'm not taking any substrings here.